lab-teddy/app/service/auth-service.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+module.exports = [
+  '$q',
+  '$log',
+  '$http',
+  '$window',
+  function($q, $log, $http, $window) {
+    $log.debug('authService');
+
+    let service = {};
+    let token = null;
+
+    function setToken(_token) {
+      $log.debug('authService.setToken()');
+
+      if(!_token) return $q.reject(new Error('No token'));
+      $window.localStorage.setItem('token', _token);
+      token = _token;
+
+      return $q.resolve(token);
+    }
+
+    service.getToken = function() {
+      $log.debug('authService.getToken()');
+
+      if(token) return $q.resolve(token);
+      token = $window.localStorage.getItem('token');
+      if(token) return $q.resolve(token);
+
+      return $q.reject(new Error('Token not found'));
+    };
+
+    service.logout = function() {
+      $log.debug('authService.logout()');
+
+      $window.localStorage.removeItem('token');
+      token = null;
+
+      return $q.resolve();
+    };
+
+    service.signup = function(user) {
+      $log.debug('authService.signup()');
+
+      // let url = `${__API_URL__}/api/signup`;
```

Comment:
I think if you try exporting this variable in your terminal first you should have access to it here so that you don't have to hard code the url each time.